osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -54,24 +115,151 @@ public override Color4 AccentColour
                     return;
                 base.AccentColour = value;
 
-                headPiece.AccentColour = value;
+                tickContainer.Children.ForEach(t => t.AccentColour = value);
+
                 bodyPiece.AccentColour = value;
-                tailPiece.AccentColour = value;
+                head.AccentColour = value;
+                tail.AccentColour = value;
             }
         }
 
         protected override void UpdateState(ArmedState state)
         {
         }
 
-        protected override void Update()
+        /// <summary>
+        /// Handles key down events on the body of the hold note.
+        /// </summary>
+        /// <param name="state">The input state.</param>
+        /// <param name="args">The key down args.</param>
+        /// <returns>Whether the key press was handled.</returns>
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            // Make sure the keypress happened within reasonable bounds of the hold note
+            if (Time.Current < HitObject.StartTime || Time.Current > HitObject.EndTime)
+                return false;
+
+            if (args.Key != Key)
+                return false;
+
+            if (args.Repeat)
+                return false;
+
+            holding = true;

```

Comment:
i'd add a comment noting the fact that `holding` can also be set to true by the head, which is how we handle input outside of the above (limited) range. helps people looking at the code to follow what's going on.